src/components/NewsletterSignup/NewsletterSignup.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Box,
+  chakra,
+  Stack,
+  useColorModeValue,
+  useStyleConfig,
+  VStack,
+} from "@chakra-ui/react";
+import React, { forwardRef } from "react";
+
+import Button from "../Button/Button";
+import Form, { FormField } from "../Form/Form";
+import Icon from "../Icons/Icon";
+import Link from "../Link/Link";
+import Text from "../Text/Text";
+import TextInput from "../TextInput/TextInput";
+import Heading from "../Heading/Heading";
+import useNYPLBreakpoints from "../../hooks/useNYPLBreakpoints";
+import { getSectionColors } from "../../helpers/getSectionColors";
+import { SectionTypes } from "../../helpers/types";
+
+interface NewsletterSignupProps {
+  /** Optional: Additional class name to add. */
+  className?: string;
+  /** Optional: Used to override the default confirmation message in the confirmation view. Accepts a string or
+   *  an element. */
+  confirmationText?: string | JSX.Element;
+  /** Optional: Appears below the title to provide details about the newsletter. Accepts a string or an element. */
+  descriptionText?: string | JSX.Element;
+  /** Optional: Appears below the input field's example text to provide any additional instructions. Accepts a string or
+   *  an element. */
+  formHelperText?: string | JSX.Element;
+  /** Optional: ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Optional: Toggles the invalid state for the email field. */
+  isInvalidEmail?: boolean;
+  /** Optional: Value to determine the section color highlight */
+  newsletterSignupType?: SectionTypes;
+  /** Required: A handler function that will be called when the form is submitted. */
+  onSubmit: (event: React.FormEvent<any>) => void;
+  /** Required: A handler function that will be called when the text input changes. */
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Optional: Used to populate the `<h3>` header title. */
+  title?: string;
+  /** Required: The value of the email text input field. */
+  valueEmail: string;
```

Comment:
I am wondering if the `valueEmail` prop could be optional. Looking at the TextInput component it seems that is optional there.

Reply:
Edwin explained the use case for this prop. Made it optional though.